### PR TITLE
Feature/habitat package

### PR DESCRIPTION
**Pull request by saghoshprogress**
### Description

This adds the habitat packages for the rails supermarket app, the sidekiq worker as well as wrapped versions of (core/) nginx, redis and postgresql with templating/defaults for supermarket.

Note some arbitrary things that may want to be modified after this PR is merged:

Naming of habitat packages
Currently supermarket and supermarket-<component> this could be supermarket-rails and the component ones may drop the prefix - this depends on the team's preference
Location of habitat packages
habitat-web for rails and habitat-sidekiq for the worker are both included in the source code directory at src/supermarket, while components remain at the top level for nginx, redis and postgresql. These may make more sense in a top-level packages directory or even in separate repositories

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by msys-sgarg** [comment]
https://buildkite.com/chef/chef-supermarket-main-habitat-build/builds/13#3f9afd43-9ef7-4a7b-8a77-b9b69bd414c0 - habitat build is passing

**Comment by msys-sgarg** [comment]
https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/368 - Adhoc build passing

File: dev-docs/habitat_configurations_for_development.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+## pg_hba.conf changes 
+Add this line in file `postgresq/config/pg_hba.conf` this is required for local development
+since we are not making postgres listen to system ip
+
+```
+host    all             all                         0.0.0.0/0    md5
+```
+
+## use the following file for ngins/config/sites_enabled/rails in dev setup
```
Comment (dheerajd-msys) [inline dev-docs/habitat_configurations_for_development.md:59]: is it typo @msys-sgarg here for nginx ?

File: nginx/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+## Usage
+
+This plan should be used by starting the service with:
+
+```
+$ hab [svc load|start] chef/supermarket-nginx --bind rails:supermarket.default
+```
+
+Note it requires a binding to the rails (web) application.
```
Comment (dheerajd-msys) [inline nginx/README.md:23]:
```suggestion
Note: It requires a binding to the rails (web) application.
```

File: nginx/README.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+The package hooks will run as the `root:root` but will make use of `worker[user|group]` configuration values (`supermarket:supermarket` by default) for the worker processes.
+
+## Bindings
+
+The application provides no exports but requires a bind to the rails web server providing the port the server is listening on, the ports nginx should listen on, the SSL switch, as well as the FQDN. Note FQDN comes in two properties as a sanitized version is required for the redirect configuration, but habitat configuration templating does not provide a way to modify values.
```
Comment (dheerajd-msys) [inline nginx/README.md:29]:
```suggestion
The application provides no exports but requires a bind to the rails web server providing the port the server is listening on, the ports nginx should listen on, the SSL switch, as well as the FQDN. Note that FQDN comes in two properties as a sanitized version is required for the redirect configuration, but habitat configuration templating does not provide a way to modify values.
```

File: dev-docs/habitat_configurations_for_development.md (hunk 1 above)
Comment (msys-sgarg) [reply dev-docs/habitat_configurations_for_development.md:59]: yes :) .. fixing it right away

**Review by msys-sgarg — COMMENTED** [review]
Looks good to me

**Review by RajeshPaul38 — APPROVED** [review]

**Pull request merged**